### PR TITLE
Sentinelone new actions - disable and enable agent

### PR DESCRIPTION
**Pull request by r7-kszczepanskagorna**
### Description

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nToken: *************5172f\nrapid7/SentinelOne:6.1.0. Step name: disable_agent\n",
    "meta": {},
    "output": {
      "affected": 0
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/sentinelone:6.1.0 --debug run < tests/disable_agent.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nToken: *************5172f\nrapid7/SentinelOne:6.1.0. Step name: disable_agent\n",
    "meta": {},
    "output": {
      "affected": 0
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/sentinelone:6.1.0 --debug run < tests/disable_agent2.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nToken: *************5172f\nrapid7/SentinelOne:6.1.0. Step name: disable_agent\n",
    "meta": {},
    "output": {
      "affected": 0
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/sentinelone:6.1.0 --debug run < tests/disable_agent3.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nToken: *************5172f\nrapid7/SentinelOne:6.1.0. Step name: disable_agent\n",
    "meta": {},
    "output": {
      "affected": 0
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/sentinelone:6.1.0 --debug run < tests/disable_agent4.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nToken: *************5172f\nrapid7/SentinelOne:6.1.0. Step name: enable_agent\n",
    "meta": {},
    "output": {
      "affected": 0
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/sentinelone:6.1.0 --debug run < tests/enable_agent.json
</summary>
</details>

<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Helpers.mk ../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 4512.195000000001ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] Validating markdown...
[SUCCESS] Passes markdown linting

[*] Validating python files for style...
./komand_sentinelone/actions/activities_types/action.py:2:1: F401 '.schema.Input' imported but unused
./komand_sentinelone/actions/agents_summary/action.py:18:19: F821 undefined name 'PluginException'
./komand_sentinelone/actions/name_available/action.py:2:1: F401 '.schema.Input' imported but unused
./komand_sentinelone/connection/connection.py:52:80: E203 whitespace before ':'
./komand_sentinelone/connection/connection.py:71:17: W503 line break before binary operator
./unit_test/test_activities_list.py:6:1: E402 module level import not at top of file
./unit_test/test_activities_list.py:7:1: E402 module level import not at top of file
./unit_test/test_activities_list.py:8:1: E402 module level import not at top of file
./unit_test/test_activities_list.py:9:1: E402 module level import not at top of file
./unit_test/test_activities_list.py:10:1: E402 module level import not at top of file
./unit_test/test_activities_list.py:38:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_activities_list.py:41:1: W293 blank line contains whitespace
./unit_test/test_activities_list.py:42:110: W291 trailing whitespace
./unit_test/test_activities_types.py:6:1: E402 module level import not at top of file
./unit_test/test_activities_types.py:7:1: E402 module level import not at top of file
./unit_test/test_activities_types.py:8:1: E402 module level import not at top of file
./unit_test/test_activities_types.py:9:1: E402 module level import not at top of file
./unit_test/test_activities_types.py:10:1: E402 module level import not at top of file
./unit_test/test_activities_types.py:38:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_activities_types.py:41:1: W293 blank line contains whitespace
./unit_test/test_activities_types.py:42:110: W291 trailing whitespace
./unit_test/test_agents_action.py:6:1: E402 module level import not at top of file
./unit_test/test_agents_action.py:7:1: E402 module level import not at top of file
./unit_test/test_agents_action.py:8:1: E402 module level import not at top of file
./unit_test/test_agents_action.py:9:1: E402 module level import not at top of file
./unit_test/test_agents_action.py:10:1: E402 module level import not at top of file
./unit_test/test_agents_action.py:11:1: E402 module level import not at top of file
./unit_test/test_agents_action.py:26:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_agents_reload.py:6:1: E402 module level import not at top of file
./unit_test/test_agents_reload.py:7:1: E402 module level import not at top of file
./unit_test/test_agents_reload.py:8:1: E402 module level import not at top of file
./unit_test/test_agents_reload.py:9:1: E402 module level import not at top of file
./unit_test/test_agents_reload.py:10:1: E402 module level import not at top of file
./unit_test/test_agents_reload.py:38:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_agents_reload.py:41:1: W293 blank line contains whitespace
./unit_test/test_agents_reload.py:42:110: W291 trailing whitespace
./unit_test/test_agents_summary.py:6:1: E402 module level import not at top of file
./unit_test/test_agents_summary.py:7:1: E402 module level import not at top of file
./unit_test/test_agents_summary.py:8:1: E402 module level import not at top of file
./unit_test/test_agents_summary.py:9:1: E402 module level import not at top of file
./unit_test/test_agents_summary.py:10:1: E402 module level import not at top of file
./unit_test/test_agents_summary.py:38:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_agents_summary.py:41:1: W293 blank line contains whitespace
./unit_test/test_agents_summary.py:42:110: W291 trailing whitespace
./unit_test/test_apps_by_agent_ids.py:6:1: E402 module level import not at top of file
./unit_test/test_apps_by_agent_ids.py:7:1: E402 module level import not at top of file
./unit_test/test_apps_by_agent_ids.py:8:1: E402 module level import not at top of file
./unit_test/test_apps_by_agent_ids.py:9:1: E402 module level import not at top of file
./unit_test/test_apps_by_agent_ids.py:10:1: E402 module level import not at top of file
./unit_test/test_apps_by_agent_ids.py:38:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_apps_by_agent_ids.py:41:1: W293 blank line contains whitespace
./unit_test/test_apps_by_agent_ids.py:42:110: W291 trailing whitespace
./unit_test/test_blacklist.py:6:1: E402 module level import not at top of file
./unit_test/test_blacklist.py:7:1: E402 module level import not at top of file
./unit_test/test_blacklist.py:8:1: E402 module level import not at top of file
./unit_test/test_blacklist.py:9:1: E402 module level import not at top of file
./unit_test/test_blacklist.py:10:1: E402 module level import not at top of file
./unit_test/test_blacklist.py:38:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_blacklist.py:41:1: W293 blank line contains whitespace
./unit_test/test_blacklist.py:42:110: W291 trailing whitespace
./unit_test/test_blacklist_by_content_hash.py:6:1: E402 module level import not at top of file
./unit_test/test_blacklist_by_content_hash.py:7:1: E402 module level import not at top of file
./unit_test/test_blacklist_by_content_hash.py:8:1: E402 module level import not at top of file
./unit_test/test_blacklist_by_content_hash.py:9:1: E402 module level import not at top of file
./unit_test/test_blacklist_by_content_hash.py:10:1: E402 module level import not at top of file
./unit_test/test_blacklist_by_content_hash.py:38:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_blacklist_by_content_hash.py:41:1: W293 blank line contains whitespace
./unit_test/test_blacklist_by_content_hash.py:42:110: W291 trailing whitespace
./unit_test/test_create_ioc_threat.py:6:1: E402 module level import not at top of file
./unit_test/test_create_ioc_threat.py:7:1: E402 module level import not at top of file
./unit_test/test_create_ioc_threat.py:8:1: E402 module level import not at top of file
./unit_test/test_create_ioc_threat.py:9:1: E402 module level import not at top of file
./unit_test/test_create_ioc_threat.py:10:1: E402 module level import not at top of file
./unit_test/test_create_ioc_threat.py:38:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_create_ioc_threat.py:41:1: W293 blank line contains whitespace
./unit_test/test_create_ioc_threat.py:42:110: W291 trailing whitespace
./unit_test/test_get_agent_details.py:6:1: E402 module level import not at top of file
./unit_test/test_get_agent_details.py:7:1: E402 module level import not at top of file
./unit_test/test_get_agent_details.py:8:1: E402 module level import not at top of file
./unit_test/test_get_agent_details.py:9:1: E402 module level import not at top of file
./unit_test/test_get_agent_details.py:10:1: E402 module level import not at top of file
./unit_test/test_get_agent_details.py:38:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_get_agent_details.py:41:1: W293 blank line contains whitespace
./unit_test/test_get_agent_details.py:42:110: W291 trailing whitespace
./unit_test/test_get_threats.py:6:1: E402 module level import not at top of file
./unit_test/test_get_threats.py:7:1: E402 module level import not at top of file
./unit_test/test_get_threats.py:8:1: E402 module level import not at top of file
./unit_test/test_get_threats.py:9:1: E402 module level import not at top of file
./unit_test/test_get_threats.py:10:1: E402 module level import not at top of file
./unit_test/test_get_threats.py:11:1: E402 module level import not at top of file
./unit_test/test_get_threats.py:12:1: E402 module level import not at top of file
./unit_test/test_get_threats.py:57:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_get_threats.py:60:1: W293 blank line contains whitespace
./unit_test/test_get_threats.py:61:110: W291 trailing whitespace
./unit_test/test_get_threat_summary.py:6:1: E402 module level import not at top of file
./unit_test/test_get_threat_summary.py:7:1: E402 module level import not at top of file
./unit_test/test_get_threat_summary.py:8:1: E402 module level import not at top of file
./unit_test/test_get_threat_summary.py:9:1: E402 module level import not at top of file
./unit_test/test_get_threat_summary.py:10:1: E402 module level import not at top of file
./unit_test/test_get_threat_summary.py:38:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_get_threat_summary.py:41:1: W293 blank line contains whitespace
./unit_test/test_get_threat_summary.py:42:110: W291 trailing whitespace
./unit_test/test_mark_as_benign.py:6:1: E402 module level import not at top of file
./unit_test/test_mark_as_benign.py:7:1: E402 module level import not at top of file
./unit_test/test_mark_as_benign.py:8:1: E402 module level import not at top of file
./unit_test/test_mark_as_benign.py:9:1: E402 module level import not at top of file
./unit_test/test_mark_as_benign.py:10:1: E402 module level import not at top of file
./unit_test/test_mark_as_benign.py:38:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_mark_as_benign.py:41:1: W293 blank line contains whitespace
./unit_test/test_mark_as_benign.py:42:110: W291 trailing whitespace
./unit_test/test_mark_as_threat.py:6:1: E402 module level import not at top of file
./unit_test/test_mark_as_threat.py:7:1: E402 module level import not at top of file
./unit_test/test_mark_as_threat.py:8:1: E402 module level import not at top of file
./unit_test/test_mark_as_threat.py:9:1: E402 module level import not at top of file
./unit_test/test_mark_as_threat.py:10:1: E402 module level import not at top of file
./unit_test/test_mark_as_threat.py:38:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_mark_as_threat.py:41:1: W293 blank line contains whitespace
./unit_test/test_mark_as_threat.py:42:110: W291 trailing whitespace
./unit_test/test_mitigate_threat.py:6:1: E402 module level import not at top of file
./unit_test/test_mitigate_threat.py:7:1: E402 module level import not at top of file
./unit_test/test_mitigate_threat.py:8:1: E402 module level import not at top of file
./unit_test/test_mitigate_threat.py:9:1: E402 module level import not at top of file
./unit_test/test_mitigate_threat.py:10:1: E402 module level import not at top of file
./unit_test/test_mitigate_threat.py:38:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_mitigate_threat.py:41:1: W293 blank line contains whitespace
./unit_test/test_mitigate_threat.py:42:110: W291 trailing whitespace
./unit_test/test_name_available.py:6:1: E402 module level import not at top of file
./unit_test/test_name_available.py:7:1: E402 module level import not at top of file
./unit_test/test_name_available.py:8:1: E402 module level import not at top of file
./unit_test/test_name_available.py:9:1: E402 module level import not at top of file
./unit_test/test_name_available.py:10:1: E402 module level import not at top of file
./unit_test/test_name_available.py:38:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_name_available.py:41:1: W293 blank line contains whitespace
./unit_test/test_name_available.py:42:110: W291 trailing whitespace
./unit_test/test_quarantine.py:6:1: E402 module level import not at top of file
./unit_test/test_quarantine.py:7:1: E402 module level import not at top of file
./unit_test/test_quarantine.py:8:1: E402 module level import not at top of file
./unit_test/test_quarantine.py:9:1: E402 module level import not at top of file
./unit_test/test_quarantine.py:10:1: E402 module level import not at top of file
./unit_test/test_quarantine.py:38:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_quarantine.py:41:1: W293 blank line contains whitespace
./unit_test/test_quarantine.py:42:110: W291 trailing whitespace
./unit_test/test_search_agents.py:6:1: E402 module level import not at top of file
./unit_test/test_search_agents.py:7:1: E402 module level import not at top of file
./unit_test/test_search_agents.py:8:1: E402 module level import not at top of file
./unit_test/test_search_agents.py:9:1: E402 module level import not at top of file
./unit_test/test_search_agents.py:10:1: E402 module level import not at top of file
./unit_test/test_search_agents.py:38:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_search_agents.py:41:1: W293 blank line contains whitespace
./unit_test/test_search_agents.py:42:110: W291 trailing whitespace
./unit_test/test_threats_fetch_file.py:6:1: E402 module level import not at top of file
./unit_test/test_threats_fetch_file.py:7:1: E402 module level import not at top of file
./unit_test/test_threats_fetch_file.py:8:1: E402 module level import not at top of file
./unit_test/test_threats_fetch_file.py:9:1: E402 module level import not at top of file
./unit_test/test_threats_fetch_file.py:10:1: E402 module level import not at top of file
./unit_test/test_threats_fetch_file.py:38:9: F841 local variable 'e' is assigned to but never used
./unit_test/test_threats_fetch_file.py:41:1: W293 blank line contains whitespace
./unit_test/test_threats_fetch_file.py:42:110: W291 trailing whitespace
[FAIL] Fails flake8 linting

[*] Validating python files for security vulnerabilities...
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.7.3
93 [0.. 50.. ]
Run started:2021-03-12 21:57:34.686745

Test results:
>> Issue: [B105:hardcoded_password_string] Possible hardcoded password: 'password'
   Severity: Low   Confidence: Medium
   Location: ./komand_sentinelone/actions/threats_fetch_file/schema.py:12
   More Info: https://bandit.readthedocs.io/en/latest/plugins/b105_hardcoded_password_string.html
11          ID = "id"
12          PASSWORD = "password"
13

--------------------------------------------------
>> Issue: [B105:hardcoded_password_string] Possible hardcoded password: ''
   Severity: Low   Confidence: Medium
   Location: ./komand_sentinelone/connection/connection.py:74
   More Info: https://bandit.readthedocs.io/en/latest/plugins/b105_hardcoded_password_string.html
73
74              token = ""
75
76              # Some consoles do not support v2.1 but some actions are not included in 2.0
77              # Instead, try getting an auth token from 2.1 first, then rollback to 2.0 if needed
78              # Find the list of supported consoles for each API in the SentinelOne API docs
79              if r.status_code != 200:

--------------------------------------------------

Code scanned:
        Total lines of code: 5999
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 2.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 2.0
                High: 0.0
Files skipped (0):
[FAIL] Fails bandit security checks

[*] Checking for typos with Misspell...
[FAIL] Fails Misspell check
help.md:95:2135: "ECT" is a misspelling of "ETC" help.md:95:5457: "TAHT" is a misspelling of "THAT" komand_sentinelone/actions/disable_agent/action.py:85:85: "ECT" is a misspelling of "ETC" komand_sentinelone/actions/disable_agent/action.py:96:37: "TAHT" is a misspelling of "THAT" komand_sentinelone/actions/disable_agent/schema.py:109:23: "ECT" is a misspelling of "ETC" komand_sentinelone/actions/disable_agent/schema.py:215:22: "TAHT" is a misspelling of "THAT" plugin.spec.yaml:1668:26: "ECT" is a misspelling of "ETC" plugin.spec.yaml:1774:25: "TAHT" is a misspelling of "THAT"

```

<summary>
make validate
</summary>
</details>

<details>

```

[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator CloudReadyConnectionCredentialTokenValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator VersionPinValidator
[*] Executing validator EncodingValidator
[*] Executing validator ExampleInputValidator
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
WARNING: URLs found that return a 4xx code. Verify they are publicly accessible and if not, update with a working URL.
violation: plugin.spec.yaml[931]: https://example.sentinelone.com
violation: plugin.spec.yaml[1385]: https://yoururl.sentinelone.net/api-doc/api-details?category=agent-actions
[*] Plugin successfully validated!

----
[*] Total time elapsed: 34228.502ms
```

<summary>
icon-validate --all .
</summary>
</details>

### CODE REVIEW

File: sentinelone/plugin.spec.yaml [hunk 1]
```diff
@@ -1580,6 +1580,74 @@ actions:
...
+        example: hostname123
+        required: false
+      reboot:
+        title: Reboot
+        description: Set true to reboot the endpoint, set false to not reboot the endpoint
+        type: boolean
+        required: true
+        example: true
+      expiration_timezone:
+        title: Expiration Timezone
```
Comment (jschipp-r7) [inline sentinelone/plugin.spec.yaml:1600]: We should do an enumerated list of timezones to make it easier for the user to choose the right string

Comment (jschipp-r7) [reply sentinelone/plugin.spec.yaml:1600]: Does the enable action not support time zones? notice it didn't have this input

File: sentinelone/komand_sentinelone/util/api.py [hunk 2]
```diff
@@ -68,3 +79,14 @@ def __get_searches(agent_details: str) -> list:
     @staticmethod
     def clean_results(results):
         return loads(dumps(results).replace("null", '"None"'))
+
+    @staticmethod
+    def __check_agents_found(agents: list) -> bool:
+        if len(agents) > 1:
+            raise PluginException(
+                cause="Multiple agents found.",
+                assistance="Please provide a unique identifier for the agent to be enable.",
```
Comment (jschipp-r7) [inline sentinelone/komand_sentinelone/util/api.py:88]:
```suggestion
                assistance="Please provide a unique agent identifier so the action can be performed on the intended agent.",
```

**Review by jschipp-r7 — CHANGES_REQUESTED** [review]
@r7-kszczepanskagorna

File: sentinelone/komand_sentinelone/actions/disable_agent/action.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import insightconnect_plugin_runtime
+from .schema import DisableAgentInput, DisableAgentOutput, Input, Output, Component
+
+
```
Comment (mrinehart-r7) [inline sentinelone/komand_sentinelone/actions/disable_agent/action.py:4]:
```suggestion
```

File: sentinelone/komand_sentinelone/actions/enable_agent/action.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import insightconnect_plugin_runtime
+from .schema import EnableAgentInput, EnableAgentOutput, Input, Output, Component
+
```
Comment (mrinehart-r7) [inline sentinelone/komand_sentinelone/actions/enable_agent/action.py:3]:
```suggestion
```

File: sentinelone/plugin.spec.yaml [hunk 5]
```diff
@@ -1580,6 +1580,74 @@ actions:
         description: Base64 encoded threat file
         type: file
         required: true
+  disable_agent:
+    title: Disable Agent
+    description: Disable agents that match the filter
```
Comment (mrinehart-r7) [inline sentinelone/plugin.spec.yaml:1585]:
```suggestion
    description: Disable agents that match a filter
```

File: sentinelone/plugin.spec.yaml [hunk 6]
```diff
@@ -1580,6 +1580,74 @@ actions:
...
+    input:
+      agent:
+        title: Agent
+        description: Agent to perform disable action on. Accepts IP address, MAC address, hostname, UUID or agent ID. Leave empty to perform action on all applicable Agents
+        type: string
+        example: hostname123
+        required: false
+      reboot:
+        title: Reboot
+        description: Set true to reboot the endpoint, set false to not reboot the endpoint
```
Comment (mrinehart-r7) [inline sentinelone/plugin.spec.yaml:1595]:
```suggestion
        description: Set true to reboot the endpoint, false to skip rebooting
```

File: sentinelone/plugin.spec.yaml [hunk 7]
```diff
@@ -1580,6 +1580,74 @@ actions:
...
```
Comment (mrinehart-r7) [inline sentinelone/plugin.spec.yaml:1635]:
```suggestion
        description: Set true to reboot the endpoint, false to skip rebooting
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

File: sentinelone/plugin.spec.yaml (hunk 1 above)
Comment (r7-kszczepanskagorna) [reply sentinelone/plugin.spec.yaml:1600]: Hi @jonschipp  :) Enable doesn't support time zones so it's only filter and reboot input.
We are waiting fore response from SentinelOne about supported timezones :)

Comment (jschipp-r7) [reply sentinelone/plugin.spec.yaml:1600]: @r7-kszczepanskagorna e-mail sent with details

File: sentinelone/help.md [hunk 8]
```diff
@@ -60,11 +140,7 @@ Example input:
 ```
 {
   "action": "connect",
-  "filter": {
-    "ids": [
-      "1000000000000000000"
-    ]
-  }
+  "filter": "{\"ids\": [\"1000000000000000000\"]}"
```
Comment (jschipp-r7) [inline sentinelone/help.md:143]:
```suggestion
    "filter": {
    "ids": [
      "1000000000000000000"
    ]
  }
```

File: sentinelone/help.md [hunk 9]
```diff
@@ -243,14 +318,6 @@ This action is used to reload an agent module (applies to Windows agents only).
 Example input:
 
 ```
```
Comment (jschipp-r7) [inline sentinelone/help.md:324]:
```suggestion
```
  "filter": {
    "ids": [
      "1000000000000000000"
    ]
  }
```

File: sentinelone/help.md [hunk 10]
```diff
@@ -242,15 +321,11 @@ This action is used to reload an agent module (applies to Windows agents only).
 
 Example input:
 
-```
-{
   "filter": {
```
Comment (jschipp-r7) [inline sentinelone/help.md:324]:
```suggestion
```
{
```

**Review by jschipp-r7 — APPROVED** [review]

**Review by jschipp-r7 — APPROVED** [review]

**Pull request merged**